cdap-app-fabric/src/main/java/io/cdap/cdap/support/job/SupportBundleJob.java

```diff
@@ -136,33 +121,72 @@ public void completeProcessing(String basePath) {
         updateFailedTask(runningTaskState.getTaskStatus(), future, basePath);
       }
     }
-    SupportBundleStatus finishBundleStatus =
-      new SupportBundleStatus(supportBundleStatus, "", CollectionState.FINISHED, System.currentTimeMillis());
+    SupportBundleStatus finishBundleStatus = SupportBundleStatus.builder(supportBundleStatus)
+      .setStatus(CollectionState.FINISHED)
+      .setFinishTimestamp(System.currentTimeMillis())
+      .build();
     addToStatus(finishBundleStatus, basePath);
   }
 
+  public void processSubTask(String taskName, String taskType, SupportBundleTask supportBundleTask, String basePath) {
```

Comment:
It's not actually subTask, you are using it for top level tasks as well. you can just have it as `executeTask`  and drop another one (call with 0 right here)